samples/embedded-map/src/components/EmbeddedMap/EmbeddedMap.tsx

```diff
@@ -1,32 +1,38 @@
-import React, { useEffect, useRef } from "react";
+import React, { useEffect, useRef, useState } from "react";
+import clsx from "clsx";
+import { Viewer, TransitionMode } from "mapillary-js";
 import {
     LayoutElement,
     LayoutElementProperties,
 } from "@vertigis/web/components";
-import { Viewer } from "mapillary-js";
+import IconButton from "@vertigis/web/ui/IconButton";
+import Sync from "@vertigis/web/ui/icons/Sync";
+import CenterMap from "@vertigis/web/ui/icons/CenterMap";
+
 // Import the necessary CSS for the Mapillary viewer to be styled correctly.
 import "mapillary-js/dist/mapillary.min.css";
 import EmbeddedMapModel from "./EmbeddedMapModel";
+import "./EmbeddedMap.css";
 
 export default function EmbeddedMap(
     props: LayoutElementProperties<EmbeddedMapModel>
 ): React.ReactElement {
     const { model } = props;
     const mlyRootEl = useRef<HTMLDivElement>();
+    const [sync, setSync] = useState(model.syncGcxMap);
+
+    const onSyncToggle = () => {
+        model.syncGcxMap = !model.syncGcxMap;
+        setSync(model.syncGcxMap);
+    };
+
+    const onRecenter = () => {
+        void model.recenter();
+    };
 
     useEffect(() => {
-        const mapillary = new Viewer(
-            mlyRootEl.current,
-            model.mapillaryKey,
-            // Mapillary node to start on.
-            "gLV8Jn5A6b6rbVRy2xhkMA",
-            {
-                component: {
-                    // Initialize the view immediately without user interaction.
-                    cover: false,
-                },
-            }
-        );
+        const mapillary = new Viewer(mlyRootEl.current, model.mapillaryKey);
+        mapillary.setTransitionMode(TransitionMode.Instantaneous);
         model.mapillary = mapillary;
 
         const handleWindowResize = () => {
```

Comment:
As mentioned back in ADO, do you mind replacing the window resize handler with `ResizeObserver` instead? We should be able to `.observe(mlyRootEl.current)`, and simply `mapillary.resize()` any time the observer is invoked.

Reply:
Done